build-scripts/manylinux-container-image/openssl-version.sh

```diff
@@ -1,6 +1,7 @@
-export OPENSSL_VERSION="1.1.1k"
-export OPENSSL_SHA256="892a0875b9872acd04a9fde79b1f943075d5ea162415de3047c327df33fbaee5"
+export OPENSSL_VERSION="openssl-3.4.1"
+export OPENSSL_SHA256="002a2d6b30b58bf4bea46c43bdd96365aaf8daa6c428782aa4feee06da197df3"
```

Comment:
3.5 version was released quite recently and will be needed for support for PQC kex (in 0.12) so unless there is something blocking the update, I would jump to that one.